dulwich/errors.py

```diff
@@ -91,6 +91,10 @@ class NotBlobError(WrongObjectException):
     type_name = "blob"
 
 
+class DirNotCleanError(Exception):
```

Comment:
Per the comment in this file, please don't add more errors here but rather near the code that raises them.

Reply:
Okay.

Reply:
[12118cc](https://github.com/jelmer/dulwich/pull/1148/commits/12118cc43385f143853c4a2306d23dc08f994aff)